py/autoevals/oai.py

```diff
@@ -100,117 +290,75 @@ def prepare_openai(client: Optional[LLMClient] = None, is_async=False, api_key=N
-    wrapped = False
     try:
-        from braintrust.oai import NamedWrapper, wrap_openai
+        openai_module = get_openai_module()
+    except Exception as e:
+        print(
+            textwrap.dedent(
+                f"""\
+            Unable to import openai: {e}
+
+            Please install it, e.g. with
+
+            pip install 'openai'
+            """
+            ),
+            file=sys.stderr,
+        )
+        raise
+
+    global warned_deprecated_api_key_base_url
+    if not warned_deprecated_api_key_base_url and (api_key is not None or base_url is not None):
+        warnings.warn(
+            "The api_key and base_url parameters are deprecated. Please use init() or call with client instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        warned_deprecated_api_key_base_url = True
 
-        if not isinstance(openai_obj, NamedWrapper):
-            openai_obj = wrap_openai(openai_obj)
+    # prepare the default openai sdk, if not provided
+    if api_key is None:
+        api_key = os.environ.get("OPENAI_API_KEY") or os.environ.get("BRAINTRUST_API_KEY")
+    if base_url is None:
+        base_url = os.environ.get("OPENAI_BASE_URL", PROXY_URL)
 
-        wrapped = True
-    except ImportError:
-        pass
-
-    if client is None:
-        # prepare the default client if not provided
-        complete_fn = None
-        rate_limit_error = None
-
-        Client = LLMClient
-
-        if is_v1:
-            client = Client(
-                openai=openai,
-                complete=openai_obj.chat.completions.create,
-                embed=openai_obj.embeddings.create,
-                moderation=openai_obj.moderations.create,
-                RateLimitError=openai.RateLimitError,
-            )
+    if hasattr(openai_module, "OpenAI"):
+        openai_module = cast(OpenAIV1Module, openai_module)
+
+        # v1 API
+        if is_async:
+            openai_obj = openai_module.AsyncOpenAI(api_key=api_key, base_url=base_url)  # type: ignore
         else:
-            rate_limit_error = openai.error.RateLimitError
-            if is_async:
-                complete_fn = openai_obj.ChatCompletion.acreate
-                embedding_fn = openai_obj.Embedding.acreate
-                moderation_fn = openai_obj.Moderation.acreate
-            else:
-                complete_fn = openai_obj.ChatCompletion.create
-                embedding_fn = openai_obj.Embedding.create
-                moderation_fn = openai_obj.Moderation.create
-            client = Client(
-                openai=openai,
-                complete=complete_fn,
-                embed=embedding_fn,
-                moderation=moderation_fn,
-                RateLimitError=rate_limit_error,
-            )
-
-    return client, wrapped
-
-
-def post_process_response(resp):
+            openai_obj = openai_module.OpenAI(api_key=api_key, base_url=base_url)  # type: ignore
+    else:
```

Comment:
based on the v0 deprecation comment in oai.py is it worth adding a runtime deprecation notice here like for the deprecated params?